gateway/gateway/service.py

```diff
@@ -22,6 +22,59 @@ class GatewayService(object):
     orders_rpc = RpcProxy('orders')
     products_rpc = RpcProxy('products')
 
+    @http(
+        "GET", "/products/<string:product_id>",
+        expected_exceptions=ProductNotFound
+    )
+    def get_product(self, request, product_id):
+        """Gets product by `product_id`
+        """
+        product = self.products_rpc.get(product_id)
+        return Response(
+            ProductSchema().dumps(product).data,
+            mimetype='application/json'
+        )
+
+    @http(
+        "POST", "/products",
+        expected_exceptions=(ValidationError, BadRequest)
+    )
+    def create_product(self, request):
+        """Create a new product - product data is posted as json
+
+        Example request ::
+
+            {
+                "id": "the_odyssey",
+                "title": "The Odyssey",
+                "passenger_capacity": 101,
+                "maximum_speed": 5,
+                "in_stock": 10
+            }
+
+
+        The response contains the new product ID in a json document ::
+
+            {"id": "the_odyssey"}
+
+        """
+
+        schema = ProductSchema(strict=True)
+
+        try:
+            # load input data through a schema (for validation)
+            # Note - this may raise `ValueError` for invalid json,
+            # or `ValidationError` if data is invalid.
+            product_data = schema.loads(request.get_data(as_text=True)).data
```

Comment:
I guess not because I see there is a test verifying this. Am confused about why it passes though...